src/App.tsx

```diff
@@ -1,61 +1,131 @@
 import './App.scss';
+import { TodoList } from './components/TodoList';
+import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import { FormEvent, useState } from 'react';
+import { User } from './types/User';
+import { Todo } from './types/Todo';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+type GetUserById = (todoId: number) => User | null;
+
+const getUserById: GetUserById = (todoId: number) => {
+  return usersFromServer.find(user => user.id === todoId) || null;
+};
+
+const getAllIds = (array: Todo[]) => {
+  const ids: number[] = [];
+
+  array.forEach(element => ids.push(element.id));
+
+  return ids;
+};
 
 export const App = () => {
+  const [todoList, setTodoList] = useState(todosFromServer);
+
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState(0);
+
+  const [titleError, setTitleError] = useState(false);
+  const [userError, setUserError] = useState(false);
+
+  const preparedTodos = todoList.map(todo => ({
+    ...todo,
+    user: getUserById(todo.userId),
+  }));
```

Comment:
[CHECKLIST-5]: The data preparation should be done in one place. In this case, the user data is being fetched in the `getUserById` function, which is not recommended. The user data should be fetched in the App component and passed down to the child components as needed.